.github/workflows/build-git-annex-macos.yaml

```diff
@@ -0,0 +1,213 @@
+          fi
+
+      - name: Run tests
+        run: |
+          case "${{ matrix.flavor }}" in
+            crippled-tmp)
+              # forces all test repos/paths into the VFAT FS
+              export TMPDIR=/Volumes/crippledfs
+              ;;
+            crippled-home)
+              export HOME=/Volumes/crippledfs
+              ;;
+            normal)
+              ;;
+            *)
+              echo "Unknown flavor ${{ matrix.flavor }}"
+              exit 1
+          esac
+
+          # Do it after we possibly setup HOME
+          git config --global user.email "test@github.land"
+          git config --global user.name "GitHub Almighty"
+
+          cd $HOME
+          export | grep -e crippledfs || :
+
+          git annex test
+
+  test-annex-more:
+    runs-on: macos-latest
+    needs: build-package
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Download git-annex package
+        uses: actions/download-artifact@v2-preview
+        with:
+          name: git-annex-macos-dmg
+
+      - name: Install git-annex package
+        run: |
+          hdiutil attach git-annex.dmg
+          rsync -a /Volumes/git-annex/git-annex.app /Applications/
+          hdiutil detach /Volumes/git-annex/
+
+      - name: Add git-annex to PATH
+        run: |
+          echo "::add-path::/Applications/git-annex.app/Contents/MacOS"
+
+      - name: Seek of dynlibs
+        run: |
+          mkdir /tmp/testrepo; cd /tmp/testrepo; git init
+          function nfailed() {
+            strace -f git-annex "$1" 2>&1 | awk "/$2.*ENOENT/{print}" | tee /dev/fd/2 | wc -l
+          }
+          # We should get some reasonable number (not 40) of directories look up for dynamic libraries
+          liblookups=
+          PS4='> '; set -x
+          test $(nfailed version "libpcre.*so") -lt 5
+          test $(nfailed init "libpcre.*so") -lt 130
+
+
+  test-datalad:
+    runs-on: macos-latest
+    needs: build-package
+    strategy:
+      matrix:
+        version: [master, maint, release]
+      fail-fast: false
+
+    steps:
+      - name: Download git-annex package
+        uses: actions/download-artifact@v2-preview
+        with:
+          name: git-annex-macos-dmg
+
+      - name: Install git-annex package
+        run: |
+          hdiutil attach git-annex.dmg
+          rsync -a /Volumes/git-annex/git-annex.app /Applications/
+          hdiutil detach /Volumes/git-annex/
+
+      - name: Add git-annex to PATH
+        run: |
+          echo "::add-path::/Applications/git-annex.app/Contents/MacOS"
+
+      - name: Install Docker
+        run: |
+          brew install docker-machine docker
+
+      - name: Set up SSH target
+        shell: bash
+        # TODO: Drop the release condition once 0.13.2 is released.
+        run: |
+          if [ "${{ matrix.version }}" != "release" ]; then
+            curl -fSsL \
+              https://raw.githubusercontent.com/datalad/datalad/master/tools/ci/prep-travis-forssh.sh \
```

Comment:
sorry - I have merged the `set -eu` in maint only for now... it will get to `master` eventually.  no need to change this up ATM.